### PR TITLE
Module shared library

### PR DESCRIPTION
**Pull request by erikbaranowski**
#### PR Description

Put shared code into a parent package for module components

#### PR Checklist

- [ ] CHANGELOG updated
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: component/module/file/file.go [hunk 1]
```diff
@@ -163,85 +123,35 @@ func (c *Component) Update(args component.Arguments) error {
...
-			c.setHealth(component.Health{
-				Health:     component.HealthTypeHealthy,
-				Message:    "module content loaded",
-				UpdateTime: time.Now(),
-			})
-		}
+		defer c.mut.RUnlock()
 
-		return err
+		_ = c.mod.LoadFlowContent(c.args.Arguments, e.(file.Exports).Content.Value)
```
Comment (mattdurham) [inline component/module/file/file.go:136]: Is there a reason we are eating this error? I noticed that it does it in the previous commit too.

File: component/module/module.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+			OnExportsChange: func(exports map[string]any) {
+				o.OnStateChange(Exports{Exports: exports})
+			},
+		}),
+	}
+}
+
+// LoadFlowContent loads the flow controller with the current component content.
+func (c *ModuleComponent) LoadFlowContent(arguments map[string]any, contentValue string) error {
+	c.mut.RLock()
```
Comment (mattdurham) [inline component/module/module.go:63]: I wonder if its clearer to wrap all this in a Rlock with defer instead of doing it in sections.

Comment (erikbaranowski) [reply component/module/module.go:63]: Good question, that's something I've been wondering about in go. Should I try to hold locks for the absolute minimal amount of time or value the clarity of the defer pattern when it's 'close enough'?

Comment (erikbaranowski) [reply component/module/module.go]: locks removed. I used to lock for the content/arguments but those are now function params after the refactor

File: component/module/file/file.go (hunk 1 above)
Comment (erikbaranowski) [reply component/module/file/file.go:136]: good question, this got me playing with it and I want to make sure the way things work between the modules is consistent. I have to update a couple things.

Comment (erikbaranowski) [reply component/module/file/file.go]: take a look at the last commit. I put comments where I hope it answers your question. I also had to add back in the content tracking and call to `LoadFlowContent` from `Update`. I had cut a little too much in my refactoring.

**Review by mattdurham — APPROVED** [review]
Awesome LGTM

**Pull request merged**